pkgs/tools/inputmethods/triggerhappy/default.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenv, fetchurl, perl }:
+
+stdenv.mkDerivation rec {
+  name = "triggerhappy-${version}";
+  version = "0.5.0";
+
+  src = fetchurl {
+    url = "https://github.com/wertarbyte/triggerhappy/archive/release/${version}.tar.gz";
+    sha256 = "af0fc196202f2d35153be401769a9ad9107b5b6387146cfa8895ae9cafad631c";
+  };
+
+  buildInputs = [ perl ];
+  installFlags = [ "DESTDIR=$(out)" ];  
+
+  postPatch = ''
+    substituteInPlace Makefile --replace "/usr/" "/"
+    substituteInPlace Makefile --replace "/sbin/" "/bin/"
+  '';
+
+  postInstall = ''
+    install -D -m 644 -t "$out/etc/triggerhappy/triggers.d" "triggerhappy.conf.examples"
+    install -D -m 644 -t "$out/usr/lib/systemd/system" "systemd/triggerhappy.service" "systemd/triggerhappy.socket"
+    install -D -m 644 -T "udev/triggerhappy-udev.rules" "$out/usr/lib/udev/rules.d/97-triggerhappy.rules"
```

Comment:
Nitpick: It would be nice to use '-t' consistently. Otherwise it can quickly get confusion which file is copied to which directory.